### ISSUE TITLE
query() can sometimes return seemingly duplicating rows 

### ISSUE THREAD

**Issue by azerum** [malthe/ts-postgres#122]
Suppose we have table 

```sql
create table ids (
  id int primary key
)
```

The query

```sql
select id from ids
```

is expected to return only distinct IDs, since `id` has unique constraint

Yet, if I have 10K rows in the table, and repeatedly run the query using 

```ts
const result = await client.query('...')
```

I sometimes get >10K rows (10001-10005). If I run `select distinct id from ids`, I also sometimes get duplicates. I noticed that it does not happen on all query runs, and happens only with many rows in table, so it might be a race condition

I reproduced this bug in vitest test (requires PostgreSQL to be installed): [azerum/ts-postgres-repro](https://github.com/azerum/ts-postgres-repro/)

The repository also includes the same test but run using `pg`. That test passes

**Comment by malthe** [comment]
Thanks for reporting this. I'll take a closer look asap, for now I have simply reproduced the error within the normal test suite.

**Comment by malthe** [comment]
Fixed in [v2.0.3](https://www.npmjs.com/package/ts-postgres/v/2.0.3).
